### PR TITLE
[PM-6296] Fix biometrics error prompt when biometrics are temporarily unavailable in browser extension (v2)

### PR DESCRIPTION
**Pull request by quexten**
## 🎟️ Tracking

https://github.com/bitwarden/clients/pull/9851
https://bitwarden.atlassian.net/browse/PM-6296

## 📔 Objective

We currently only have two ways of "detecting biometric availability", one is osSupported which is static per OS on browser, and dynamically detects availability on mac. The other way is biometrics failing during unlock.

Since biometrics can become temporarily unavailable (f.e closing the lid on a mac disables touchid), we need a way to dynamically detect availability in the browser extension.

This PR extends the IPC protocol with an availability check message, and the extension checks for availability before triggering the prompt. If this is due to a manual, intentional user action, a warning dialog is shown. If this is due to automatic "touchid on extension open", then no warning is shown.

### v2:
Rebase and fixed. Caused an issue building main so had to revert the first PR in https://github.com/bitwarden/clients/pull/10373

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by jlf0dev — DISMISSED** [review]
🤞

File: apps/browser/src/platform/services/background-browser-biometrics..ts
Comment (dani-garcia) [inline apps/browser/src/platform/services/background-browser-biometrics..ts:1]: I don't know if GitHub is being weird, but it looks like this file has two periods on the file extension:
```
apps/browser/src/platform/services/background-browser-biometrics..ts
                                                                 ^
```

**Review by dani-garcia — APPROVED** [review]
LGTM, just a small thing I noticed in one of the file names, but not sure if it's just GitHub being weird.

File: apps/browser/src/platform/services/background-browser-biometrics..ts
Comment (quexten) [reply apps/browser/src/platform/services/background-browser-biometrics..ts:1]: Huh, not sure how that snuck in. Fixed in the rebased version.

**Comment by quexten** [comment]
Had to rebase for linux biometrics since the biometrics interface changed quite a lot.

**Review by dani-garcia — DISMISSED** [review]

File: apps/browser/src/platform/services/background-browser-biometrics.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import { NativeMessagingBackground } from "../../background/nativeMessaging.background";
+
+import { BrowserBiometricsService } from "./browser-biometrics.service";
+
+@Injectable()
+export class BackgroundBrowserBiometricsService extends BrowserBiometricsService {
+  constructor(private nativeMessagingBackground: NativeMessagingBackground) {
+    super();
+  }
```
Comment (jlf0dev) [inline apps/browser/src/platform/services/background-browser-biometrics.ts:11]: Just as info, another way to handle this is using `inject`.
```suggestion
    const nativeMessagingBackground = inject(NativeMessagingBackground);
```

**Review by jlf0dev — DISMISSED** [review]
🚀

**Review by dani-garcia — APPROVED** [review]

**Review by dani-garcia — APPROVED** [review]

**Review by jlf0dev — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Pull request merged**